High resolution joystick

ejaquay opened the pull request:
Lots of changes to get this working.  Please someone else test this too!.    CoCo Max is not done yet, but should be trivial once I find something to test is with,   Also physical joysticks have not been tested because I have none.

Commit:
Move joystick code from keyboard to joystickinput

Commit:
Return high rez mouse values from Vcc.c

joystick(x,y) function now returns high resolution data from the
mouse. (which the rest of code still ignores) The high byte of
x,y is the 6 bit low resolution value and the low byte of x,y
provides the higher resolution.

Also renamed Right and Left variables in config.c to RightJS
and LeftJS to preserve maintainer sanity,

Commit:
Joystick software and tandy high res working

Remaining: enable high resolution joystick config dialog

Commit:
Keyboard config connected to hires joystick code

Tandy high resolution and software high resolution
joysticks both seem to be working with mouse.

Added new radio button for software high resolution.

Disabled system mouse cursor while in client area.

TODO:
 o Refine software hires table
 o Add code to set ramp values for and enable CC-MAX
 o Add checkbox to show/hide system mouse cursor
 o Test with real analog joystick attached to pc (Have none)
 o Lots of testing

Commit:
Optionally remove mouse pointer
User can now enable / disable mouse pointer from joystick config

BGPierce commented:
I'll try to check it out in the next couple of days.
Thanks,
Bill P.
 
 
 
-----Original Message-----
From: Ed Jaquay ***@***.***>
To: VCCE/VCC ***@***.***>
Cc: Subscribed ***@***.***>
Sent: Sun, Apr 10, 2022 5:44 pm
Subject: [VCCE/VCC] High resolution joystick (PR #95)

Lots of changes to get this working. Please someone else test this too!. CoCo Max is not done yet, but should be trivial once I find something to test is with, Also physical joysticks have not been tested because I have none.
You can view, comment on, or merge this pull request online at:
  https://github.com/VCCE/VCC/pull/95
Commit Summary
   
   - 36c46e9 Move joystick code from keyboard to joystickinput
   - 0555f38 Return high rez mouse values from Vcc.c
   - bbf4d50 Joystick software and tandy high res working
   - 8c1e5bc Keyboard config connected to hires joystick code
   - b846589 Optionally remove mouse pointer

File Changes 
 (14 files)   
   -  M Vcc.c (66) 
   -  M Vcc.rc (22) 
   -  M config.c (304) 
   -  M config.h (3) 
   -  M defines.h (1) 
   -  M hd6309.c (8) 
   -  M joystickinput.c (535) 
   -  M joystickinput.h (35) 
   -  M keyboard.c (468) 
   -  M keyboard.h (30) 
   -  M mc6809.c (13) 
   -  M mc6821.c (155) 
   -  M mc6821.h (3) 
   -  M resource.h (3) 

Patch Links:
   
   - https://github.com/VCCE/VCC/pull/95.patch
   - https://github.com/VCCE/VCC/pull/95.diff
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

Commit:
Tweaks to tandy ramp values and Code Pages III fix

Fixed cursor bounce right issue with Code Pages III
Enabled CCMAX checkboxes, but CCMAX is not yet working.

Commit:
CCMax hires type joystick now working

Tested with CM3 version 3.1 seems good

Not yet tested: Physical analog joystick attached to PC
I Don't Have one.

Commit:
Fix bug software hires joystick

Also Renamed DAC_Clock global to Ramp_Clock

Commit:
Added check on Tandy Hires ramp start

Commit:
Modest improvement to software high res joystick

The best vcc is capable of doing using the current approach
using software is about 5x resolution.  A real CoCo is said
to beable to do 10x.

Commit:
Adjust software high res JoyStick to 5x

Because Vcc does not know timing difference between of LDB ,X (4 cycles)
and LDB $FF00 (5 cycles) at the time of the DAC compare (when $FFOO is
read) it can at best get 5x resolution instead of 10x for real CoCo with
published high resolution joystick driver.  This change adjusts DAC rising
and falling tables to suit.

Commit:
Move mouse pointer to main window at dialog exit

Commit:
Only move mouse to corner if not in main window

Last fix moved mouse to upper left corner of Vcc window when a
click was used to regain keyboard focus.  This patch leaves mouse
pointer at the click point when this is the case.
